low-level/framework/utils/os_utils.py

```diff
@@ -29,3 +31,28 @@ class OSUtils:
     def get_fqdn():
         """Get fully qualified domain name for current node."""
         return socket.getfqdn()
+
+    @staticmethod
+    def get_host_type():
+        """
+        Determine and get host hba type using lspci.
+
+        Returns:
+            scsi_host or fc_host if HBA card is installed
+            Otherwise None.
+        """
+        host_type = ""
+
+        # Check for fc host
+        fc_host_check_cmd = "lspci | grep -Ei 'Fibre Channel'"
+        _, _, rc = PipedProcess(fc_host_check_cmd).run()
+        if rc == 0:
+            host_type = "fc_host"
+
+        # check for scsi host
+        sas_hba_check_cmd = "lspci | grep -Ei 'Serial Attached SCSI|SAS|LSI|SCSI storage controller'"
+        _, _, rc = PipedProcess(sas_hba_check_cmd).run()
```

Comment:
Did it required to grep  substring "storage controller" as well because it may not in output ?

Reply:
There are many vendors for providing HBA card. RedHat VM is having such type.
```
[root@ssc-vm-rhev4-1125 ~]# lspci | grep 'SCSI storage controller'
00:08.0 SCSI storage controller: Red Hat, Inc. Virtio SCSI
```